### ISSUE TITLE
Staking v2

### ISSUE THREAD

**Issue by tolbrino** [hoprnet/hopr-devrel#138]
The current `stake.hoprnet.org` is meant to be updated for the next season of staking. 

# Specification
1. Deployment: new smart contract and new staking UI on stake.hoprnet.org which points users to stake-s1.hoprnet.org for managing the old program
2. Migration: Manual unstake of NFTs and HOPR tokens on stake-s1.hoprnet.org, manual stake in the new program on stake.hoprnet.org.
*Note*: There’s a lot of things that can go wrong with automated re-staking of funds and NFTs and wrapping we do not even try automating that with a single button click and instead keep the two steps entirely separate.
-> *TODO*: create task for Rich McDowell to describe un-stake/stake
3. Smart Contracts: Identical to previous version with the following changes:
3.1.  Removing redeem deadlines
Note: we introduced deadlines as a sybil prevention mechanism but we did not observe attempts in that direction so we might as well remove them to keep the design simple. Otherwise we’d need to update the NFT redeem deadlines (impossible) or provide a mapping of old redeem deadline => new redeem deadline which is significant extra work.
3.2 Exposing getter endpoints for Whitelist management
3.2.1 `isNftTypeAndRankRedeemed1(string memory nftType, string memory nftRank, address hodler) external view returns (bool)`
3.2.2 `isNftTypeAndRankRedeemed2(uint256 nftTypeIndex, string memory nftRank, address hodler) external view returns (bool)`
3.2.3 `isNftTypeAndRankRedeemed3(uint256 nftTypeIndex, uint256 boostNumerator, address hodler) external view returns (bool)`
3.2.4 `isNftTypeAndRankRedeemed4(string memory nftType, uint256 boostNumerator, address hodler) external view returns (bool)`
3.2.5 `StakedHoprTokens(address hodler) returns uint256`
4. UI: Same as before with the following changes:
4.1 A link on top of the page that takes users to stake-s1.hoprnet.org with the text “Unstake your season1 stake by following the link”
4.2 Remove all redeem deadlines from UI and logic.
4.3 Add unlock ability on both stake versions

# Definition of Done
- [x] Staking rules and procedure are clearly documented on the website
- [x] A user can claim their stake from the first staking season
- [x] A user can stake on the new site
- [x] Old website is accessible at stake-s1.hoprnet.org
- [x] New website is accessible at stake.hoprnet.org
- [x] Links between season1 and new site are in place
- [x] Create subgraph for the season2
- [x] All existing assets (NFT images) are carried to season 2 FE.
- [x] Deploy SC on xDAI and deploy the corresponding xDAI subgraph
- [x] Verify SC

# Blockers

- https://github.com/hoprnet/hopr-devrel/pull/152
- https://github.com/hoprnet/hopr-stake/issues/42
- https://github.com/hoprnet/hopr-stake/issues/46
- https://github.com/hoprnet/hopr-stake/pull/43

**Comment by nionis** [comment]
Short talk with Rich, we would like to also improve:

1. better UX when u are on the wrong network & not connected to metamask
2. over time, website slows down, would be nice to take a look

**Comment by tolbrino** [comment]
> Short talk with Rich, we would like to also improve:
> 
>     1. better UX when u are on the wrong network & not connected to metamask
> 
>     2. over time, website slows down, would be nice to take a look

@SCBuergel Can you comment?

**Comment by SCBuergel** [comment]
> > Short talk with Rich, we would like to also improve:
> > ```
> > 1. better UX when u are on the wrong network & not connected to metamask
> > 
> > 2. over time, website slows down, would be nice to take a look
> > ```
> 
> @SCBuergel Can you comment?

IMO we're getting into feature creep - it's working well enough right now. If we can make a significant improvement by spending 1-2h on it, let's do it but not more than that.

**Comment by robertkiel** [comment]
The requested changes seem to remove some very nasty behavior. Hence we might save a lot of time fixing both things now - rather than later.

**Comment by SCBuergel** [comment]
Staking is a procut that does not need to be continued or continuously developed. It does not need to get better than "working" and it just needs to work good enough. If we didn't encounter critical errors in the 6 months until now, it's unlikely we will encounter them in the months to come.

**Comment by jjperezaguinaga** [comment]
An important piece of information to be aware of is that both the `unlock` logic for the locked `xHOPR` tokens and `unstake` logic for the staked NFTs are not yet implemented. They are both, however, done via the same contract call, making this implementation trivial.

This means that in this particular component,

https://github.com/hoprnet/hopr-devrel/blob/d2195cdc05dfe040eec36db53a481ed8154af0fa/stake/packages/frontend/components/StakeXHoprTokens.tsx#L281-L293

needs to have an `onClick` handler that calls [this](https://github.com/hoprnet/hopr-stake/blob/9ade7a108235df16ca10d0ec01a569238d156441/contracts/HoprStake.sol#L236) particular method with the user's address. This function will both return the locked `xHOPR` tokens and return the staked NFTs.

**Comment by nionis** [comment]
1. `Staking rules and procedure are clearly documented on the website` - I suspect with this u mean S2 website?
2. I tried `season1.stake.hoprnet.org` but it seems that chrome considers that a malicious URL ([example](https://serverfault.com/questions/559537/wildcard-certificate-causes-warning-on-google-chrome-only)), can we use `stake-s1.hoprnet.org` ?
3. `unlock` was not implemented and needs to be added in scope [PR](#142)

**Comment by tolbrino** [comment]
> 1. `Staking rules and procedure are clearly documented on the website` - I suspect with this u mean S2 website?
> 
yes
>     2. I tried `season1.stake.hoprnet.org` but it seems that chrome considers that a malicious URL ([example](https://serverfault.com/questions/559537/wildcard-certificate-causes-warning-on-google-chrome-only)), can we use `stake-s1.hoprnet.org` ?
This seems to be a SSL setup problem. However, I cannot reach that site right now.
> 
>     3. `unlock` was not implemented and needs to be added in scope [PR](#142)
Looks ok to me. Please add to the epic. \cc @SCBuergel

**Comment by SCBuergel** [comment]
I don't have an account to test but do you mean unstaking has never been implemented? If that's the case then, sure, we need that no matter what. In the description that was assumed to be there already.

**Comment by nionis** [comment]
Epic updated

**Comment by nionis** [comment]
An update on my end, other than #145 and #142, waiting on the new SC to start updating stake-v2 website.

**Comment by nionis** [comment]
Merged #142 #145 #147, I think the new SC is now almost finalized, so I will look for differences in the ABI and update stake-v2 website.

**Comment by QYuQianchen** [comment]
Stake S2 deployed at [0x2cdd13ddb0346e0f620c8e5826da5d7230341c6e](https://blockscout.com/xdai/mainnet/address/0x2cDD13ddB0346E0F620C8E5826Da5d7230341c6E/transactions)
Subgraph endpoints for Staking S2
- Goerli development: https://api.thegraph.com/subgraphs/name/hoprnet/staking-season2-development
- xDAI production: https://api.thegraph.com/subgraphs/name/hoprnet/staking-season2

**Comment by nionis** [comment]
Started testing S2 website. @tolbrino did we get ACK on the Sunday launch for S2?

**Comment by nionis** [comment]
S2 website is live, testing various interactions today.

**Comment by nionis** [comment]
S2 websites tested.

**Comment by nionis** [comment]
This epic may close, @QYuQianchen what do u think?

**Comment by QYuQianchen** [comment]
ACK. Only got `Failed to load resource: the server responded with a status of 415 ()` when loading Stake-S1 website, but it's probably related to proxy providers but not the UI, and more specifically not stake-s1 UI.

**Comment by nionis** [comment]
Okay, opening an issue for that one.

**Comment by tolbrino** [comment]
Re-opening until the blocking bug has been resolved.

**Comment by QYuQianchen** [comment]
Should we pre-open a new issue for actions to be taken when S2 approaches its end?

**Comment by tolbrino** [comment]
> Should we pre-open a new issue for actions to be taken when S2 approaches its end?

Good call.

I created https://github.com/hoprnet/hopr-devrel/issues/163
